src/status_im/contexts/wallet/wallet_connect/effects.cljs

```diff
@@ -41,10 +46,47 @@
  :effects.wallet-connect/approve-session
  (fn [{:keys [web3-wallet proposal supported-namespaces on-success on-fail]}]
    (let [{:keys [params id]} proposal
-         approved-namespaces (wallet-connect/build-approved-namespaces params
-                                                                       supported-namespaces)]
+         approved-namespaces (wallet-connect/build-approved-namespaces
+                              params
+                              supported-namespaces)]
      (-> (.approveSession web3-wallet
                           (clj->js {:id         id
                                     :namespaces approved-namespaces}))
          (promesa/then on-success)
          (promesa/catch on-fail)))))
+
+(rf/reg-fx
+ :effects.wallet-connect/sign-message
+ (fn [{:keys [password address data on-success on-fail]}]
+   (-> (promesa/-> {:data     data
```

Comment:
The nesting for `->` and `p/->` feels odd. Is this the only way to achieve this effect? 
Did you consider `p/let` for nested promises?

Reply:
I was wondering the same thing, whether there's a way to 'p/->' and 'p/then' with 'p/catch' on it without nesting it under ->. Maybe @ilmotta knows?

Reply:
also, i think even with `p/let` it would have to be wrapped in `(-> ...)` to be able to `catch` it

Reply:
removed p/-> altogether cause it's not needed. Thanks 👍